app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulation.java

```diff
@@ -0,0 +1,169 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import java.util.Objects;
+import java.util.Set;
+
+import org.sagebionetworks.bridge.json.BridgeTypeName;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshalling;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.Sets;
+
+/**
+ * A sub-population of the study participants who will receive a unique consent based on selection by a data group, 
+ * an application version, or both. 
+ */
+@DynamoDBTable(tableName = "Subpopulation")
+@BridgeTypeName("Subpopulation")
+public final class DynamoSubpopulation implements Subpopulation {
+
+    private String studyIdentifier;
+    private String guid;
+    private String name;
+    private String description;
+    private boolean required;
+    private boolean deleted;
+    private Integer minAppVersion;
+    private Integer maxAppVersion;
+    private Long version;
+    private Set<String> allOfGroups;
+    private Set<String> noneOfGroups;
+
+    public DynamoSubpopulation() {
+        this.allOfGroups = Sets.newHashSet();
+        this.noneOfGroups = Sets.newHashSet();
+    }
+    
+    @JsonIgnore
+    @Override
+    @DynamoDBHashKey
+    public String getStudyIdentifier() {
+        return studyIdentifier;
+    }
+    @Override
+    public void setStudyIdentifier(String studyIdentifier) {
+        this.studyIdentifier = studyIdentifier;
+    }
+    @Override
+    @DynamoDBRangeKey
+    public String getGuid() {
+        return guid;
+    }
```

Comment:
The two operations that are common here are 1) select all sub pops for a study, and 2) select an individual sub pop. It seems the hashKey: study, range: guid approach does these both without any indices, which incur overhead. My understanding is that DDB handles hash/range queries well.

Regarding GUID vs. name, researchers will definitely interact with these by name, but internally it is much easier for us if we have an identifier we know to be unique. So this object has both a name and a guid, the name is editable even after creation, the guid is not.